### PR TITLE
memberships: modal for errors

### PR DESCRIPTION
**Pull request by DTabidze**

### CODE REVIEW

File: core/auth/memberships/index.html [hunk 1]
```diff
@@ -56,5 +56,17 @@ <h4>Transitive Groups</h4>
...
+        <article>
+            <h2>Error:</h2>
+            <p id="error-message-content">{{ .ErrorMessage }}</p>
+            <footer>
+                <button id="error-cancel-button" class="secondary error-cancel-button">Cancel</button>
+            </footer>
+        </article>
+    </dialog>
+    {{ end }}
+    <script src="/static/main.js"></script>
```
Comment (giolekva) [inline core/auth/memberships/index.html:70]: This and some other parts of this file are duplicated in group.html as well. To avoid that you can use template inheritance. Read about `define`/`block` segments in go templates. Also check how inheritance is used by env-manager: `core/installer/welcome/env-manager-tmpl` folder.

File: core/auth/memberships/main.go [hunk 2]
```diff
@@ -551,6 +553,10 @@ type GroupData struct {
 	Membership string
 }
 
+type ErrorResponse struct {
+	Message string `json:"message"`
+}
+
```
Comment (giolekva) [inline core/auth/memberships/main.go:559]: This does not look to be used

**Review by giolekva — CHANGES_REQUESTED** [review]

File: core/auth/memberships/index.html (hunk 1 above)
Comment (DTabidze) [reply core/auth/memberships/index.html]: Check it.

File: core/auth/memberships/main.go (hunk 2 above)
Comment (DTabidze) [reply core/auth/memberships/main.go]: Done.

File: core/auth/memberships/main.go [hunk 3]
```diff
@@ -563,6 +561,34 @@ func (s *Server) checkIsOwner(w http.ResponseWriter, user, group string) (bool,
 	return true, nil
 }
 
+type templates struct {
+	group *template.Template
+	index *template.Template
```
Comment (giolekva) [inline core/auth/memberships/main.go:566]: `index` is used to render user page, let's rename it to `user`

File: core/auth/memberships/memberships-tmpl/group.html [hunk 4]
```diff
@@ -1,14 +1,8 @@
...
-<html lang="en" data-theme="light">
+{{ define "content" }}
+{{- $parentGroupName := .GroupName }}
 <head>
-    <meta charset="UTF-8">
-    <meta name="viewport" content="width=device-width, initial-scale=1.0">
-    <title>Group Membership Managment</title>
-    <link rel="stylesheet" href="/static/pico.2.0.6.min.css">
-    <link rel="stylesheet" href="/static/main.css">
+    <title>Group Management</title>
```
Comment (giolekva) [inline core/auth/memberships/memberships-tmpl/group.html:4]: Title should be configurable as well, there should be a block for it in the base.html

With current implementation `<head>` tag will end up inside `<body>` which is not right

File: core/auth/memberships/memberships-tmpl/group.html [hunk 5]
```diff
@@ -110,6 +104,6 @@ <h2>Confirm Your Action</h2>
             </footer>
         </article>
     </dialog>
-    <script src="/static/main.js"></script>
-</body>
-</html>
+    <link rel="stylesheet" href="/static/main.css">
```
Comment (giolekva) [inline core/auth/memberships/memberships-tmpl/group.html:107]: Should not css go in to the `base.html` as well?

File: core/auth/memberships/memberships-tmpl/index.html
Comment (giolekva) [inline core/auth/memberships/memberships-tmpl/index.html:1]: This also should be named `user.html`

File: core/auth/memberships/memberships-tmpl/group.html [hunk 6]
```diff
@@ -110,6 +104,6 @@ <h2>Confirm Your Action</h2>
...
+{{ end }}
+{{- template "base.html" . -}}
```
Comment (giolekva) [inline core/auth/memberships/memberships-tmpl/group.html:109]: Why is this needed?

**Review by giolekva — CHANGES_REQUESTED** [review]

File: core/auth/memberships/memberships-tmpl/group.html (hunk 4 above)
Comment (DTabidze) [reply core/auth/memberships/memberships-tmpl/group.html]: Done.

File: core/auth/memberships/memberships-tmpl/group.html (hunk 5 above)
Comment (DTabidze) [reply core/auth/memberships/memberships-tmpl/group.html]: Done.

File: core/auth/memberships/memberships-tmpl/group.html (hunk 6 above)
Comment (DTabidze) [reply core/auth/memberships/memberships-tmpl/group.html]: removed.

File: core/auth/memberships/memberships-tmpl/index.html
Comment (DTabidze) [reply core/auth/memberships/memberships-tmpl/index.html:1]: Done.

File: core/auth/memberships/memberships-tmpl/base.html [hunk 7]
```diff
@@ -3,7 +3,11 @@
 <head>
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>{{ .Title }}</title>
     <link rel="stylesheet" href="/static/pico.2.0.6.min.css">
+    {{if .AdditionalCSS }}
+        <link rel="stylesheet" href="/static/main.css">
+    {{end}}
```
Comment (giolekva) [inline core/auth/memberships/memberships-tmpl/base.html:10]: Why is this being loaded conditionally? Should not it be applied to all pages?

File: core/auth/memberships/memberships-tmpl/base.html [hunk 8]
```diff
@@ -3,7 +3,11 @@
 <head>
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>{{ .Title }}</title>
```
Comment (giolekva) [inline core/auth/memberships/memberships-tmpl/base.html:6]: Title should not be coming from go, define it as a block and provide value from `group.html` and `user.html` files. See how it is done in `core/auth/ui/templates/`

Comment (giolekva) [reply core/auth/memberships/memberships-tmpl/base.html:6]: Does it break something on user page?

**Review by giolekva — CHANGES_REQUESTED** [review]

File: core/auth/memberships/memberships-tmpl/group.html [hunk 9]
```diff
@@ -1,3 +1,6 @@
+{{ define "title" }}
+    Group Management - {{ .GroupName }}
```
Comment (giolekva) [inline core/auth/memberships/memberships-tmpl/group.html:2]: Remove `Management`

File: core/auth/memberships/memberships-tmpl/user.html [hunk 10]
```diff
@@ -1,3 +1,6 @@
+{{ define "title" }}
+    User Management - {{ .CurrentUser }}
```
Comment (giolekva) [inline core/auth/memberships/memberships-tmpl/user.html:2]: Remove `Management`

**Review by giolekva — APPROVED** [review]
Feel free to merge after addressing two minor comments
